pkgs/servers/monitoring/cockpit/default.nix

```diff
@@ -0,0 +1,168 @@
+    docbook_xml_dtd_43
+    docbook_xsl
+    findutils
+    gettext
+    git
+    glib.dev
+    libxcrypt
+    libxslt.bin
+    nodejs
+    pam.out
+    pkg-config
+    python3Packages.python
+    python3Packages.setuptools
+    systemd
+    ripgrep
+    xmlto
+  ];
+
+  buildInputs = [
+    glib
+    gnutls
+    json-glib
+    krb5
+    libssh
+    polkit
+    udev
+  ];
+
+  patches = [
+    # Instead of requiring Internet access to do an npm install to generate the package-lock.json
+    # it copies the package-lock.json already present in the node_modules folder fetched as a git
+    # submodule.
+    ./nerf-node-modules.patch
+
+    # sysconfdir is $(prefix)/etc by default and it breaks the configuration file loading feature
+    # changing sysconfdir to /etc breaks the build in this part of the makefile because it tries
+    # to write to /etc inside the sandbox
+    # this patch redirects it to write to $out/etc instead of /etc
+    ./fix-makefiles.patch
+  ];
+
+  # preConfigurePhases = [ "preConfigure" ];
+  postPatch = ''
+    patchShebangs \
+      tools/escape-to-c \
+      tools/make-compile-commands \
+      tools/node-modules \
+      tools/termschutz \
+      tools/webpack-make \
+      test/common/tap-cdp \
+      test/common/pixel-tests \
+      test/common/run-tests
+
+    export HOME=$(mktemp -d)
+
+    cp node_modules/.package-lock.json package-lock.json
+
+    substituteInPlace src/systemd_ctypes/libsystemd.py \
+      --replace libsystemd.so.0 ${systemd}/lib/libsystemd.so.0
+
+    substituteInPlace pkg/**/*.js pkg/**/*.jsx \
+      --replace '"/usr/bin/' '"' \
+      --replace '"/bin/' '"'
+
+    # for patch testing
+    # cat pkg/users/local.js
+
+    echo "m4_define(VERSION_NUMBER, [${version}])" > version.m4
+  '';
+
+  configureFlags = [
+    "--enable-prefix-only=yes"
+    "--sysconfdir=/etc"
+    "--disable-pcp" # TODO: figure out how to package its dependency
+    "--with-default-session-path=/run/wrappers/bin:/run/current-system/sw/bin"
+  ];
+
+  enableParallelBuilding = true;
+
+  fixupPhase = ''
+    runHook preFixup
+
+    patchShebangs \
+      $out/libexec/cockpit-certificate-helper \
+      $out/share/cockpit/motd/update-motd \
+      $out/libexec/cockpit-desktop
+
+    PATH=${python3Packages.python.withPackages (p: with p;[ pygobject3 ])}/bin patchShebangs $out/libexec/cockpit-client
+
+    wrapProgram $out/libexec/cockpit-certificate-helper \
+      --prefix PATH : ${lib.makeBinPath [ coreutils openssl ]} \
+      --run 'cd $(mktemp -d)'
+
+    wrapProgram $out/share/cockpit/motd/update-motd \
+      --prefix PATH : ${lib.makeBinPath [ gnused ]}
+
+    substituteInPlace $out/libexec/cockpit-desktop \
+      --replace ' /bin/bash' ' ${bash}/bin/bash'
+
+    substituteInPlace $out/share/polkit-1/actions/org.cockpit-project.cockpit-bridge.policy \
+      --replace /usr $out
+
+    runHook postFixup
+  '';
+
+  doCheck = true;
+  checkInputs = [ python3Packages.pytest ];
+  checkPhase = "make pytest";
```

Comment:
Why not execute pytest directly?

Reply:
I added more tests from upstream to checkPhase

I wanted to follow the upstream workflow, so stuff can be more resilient if they change something. At least that is the intention.